### PR TITLE
#13 Adding authorization via Devise gem

### PR DESCRIPTION
**Pull request by TarasBardiuk**
- [x] @Kmalynss 
- [x] @AndriySolonyna
- [x] @f1nwe 
Adding authorization via Devise gem by login-password. Profile registration, editing etc. disabled, first test user created in seeds.rb. Created home-controller with index view (for custom root page).
**UPD.** Code updated according to comments. Added tests for email validation.

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ AllCops:
     - 'tmp/**/*'
     - 'config/initializers/new_framework_defaults.rb'
     - 'bin/*'
+    - 'db/schema.rb'
+    - 'db/migrate/20161217105518_devise_create_users.rb'
```
Comment (AndriySolonyna) [inline .rubocop.yml]: please remove migration from ignore
What kind of errors you have in 20161217105518_devise_create_users.rb ?

File: app/controllers/home_controller.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+class HomeController < ApplicationController
+  skip_before_action :authenticate_user!, only: [:index]
+
+  def index
+    # index method
```
Comment (AndriySolonyna) [inline app/controllers/home_controller.rb]: remove this comment

File: app/helpers/home_helper.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# frozen_string_literal: true
```
Comment (AndriySolonyna) [inline app/helpers/home_helper.rb]: remove all autogenerated files if you don't use them

File: db/seeds.rb [hunk 4]
```diff
@@ -9,3 +9,9 @@
 #   movies = Movie.create([{ name: 'Star Wars' },
 #   { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+# Initialize first account:
+User.create! do |u|
```
Comment (AndriySolonyna) [inline db/seeds.rb]: write it in 1 line
User.create!(email: 'sdfdsf@email.com', password: '123456', password_confirmation: '123456')

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable,
+         :rememberable, :trackable
```
Comment (AndriySolonyna) [inline app/models/user.rb]: add validation for email

File: app/views/users/sessions/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  <% if devise_mapping.rememberable? -%>
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end -%>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
```
Comment (AndriySolonyna) [inline app/views/users/sessions/new.html.erb]: add disable_with to  submit button 
http://api.rubyonrails.org/classes/ActionView/Helpers/FormTagHelper.html

File: test/fixtures/users.yml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (AndriySolonyna) [inline test/fixtures/users.yml]: remove this file
We will use factory_girl gem for creating test data

**Comment by AndriySolonyna** [comment]
please rebase your branch with master

**Comment by Kmalynss** [comment]
@TarasBardiuk please squash commits ant then merge

File: app/policies/application_policy.rb [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    @user = user
+    @record = record
+  end
+
+  def index?
+    false
+  end
+
+  def show?
+    scope.where(id: record.id).exists?
```
Comment (igorkasyanchuk) [inline app/policies/application_policy.rb]: if you have "record.id" probably it's already means that object exists?

File: Gemfile [hunk 9]
```diff
@@ -1,56 +1,36 @@
...
+gem 'coffee-rails', '~> 4.2'
+gem 'devise'
+gem 'jbuilder', '~> 2.5'
+gem 'jquery-rails'
+gem 'omniauth'
+gem 'omniauth-facebook'
 gem 'puma', '~> 3.0'
-# Use SCSS for stylesheets
+gem 'pundit'
+gem 'rails', '~> 5.0.0', '>= 5.0.0.1'
```
Comment (igorkasyanchuk) [inline Gemfile]: specify only one version of rails

**Pull request merged**